crates/eframe/src/epi.rs

```diff
@@ -668,10 +671,69 @@ impl Frame {
     }
 
     /// A place where you can store custom data in a way that persists when you restart the app.
+    #[cfg(not(target_arch = "wasm32"))]
     pub fn storage(&self) -> Option<&dyn Storage> {
         self.storage.as_deref()
     }
 
+    /// Request the current frame's pixel data. Needs to be retrieved by calling [`Frame::screenshot`]
+    /// during [`App::post_rendering`].
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
What is the limitation for implementing this on web?

Reply:
Mostly my inexperience with wasm and any kind of web-dev. I took a stab at trying to port the screenshot example to wasm with the eframe-template though, and it seems like App::post_rendering is never called on the web?
From the backend perspectiv, 3 things are required;

1. A flag needs to be set during App::update.
2. The flag needs to be read during the backend's paint call, such that the data can be pulled from the GPU
3. The data needs to be delivered back to the app during App::post_rendering

I chose to use the Frame struct for delivering the flag and image data between the app and the backend, which was easy on native since everything could be handled inside the WinitApp::paint function. I didn't look too closely at the architecture of the webpainter yet, but as App::post_rendering is never called, there would be no way to deliver the image data back to the app.

If this is solved, I believe it would be possible to more or less copy-paste the actual screen capturing logic for wgpu and glow.